wdl-doc/src/callable.rs

```diff
@@ -0,0 +1,292 @@
+    fn render_group_inputs(&self) -> Markup {
+        let group_tables = self
+            .input_groups()
+            .into_iter()
+            .filter(|group| *group != "Common")
+            .map(|group| {
+                html! {
+                    h3 { (group) }
+                    table class="border" {
+                        thead class="border" { tr {
+                            th { "Name" }
+                            th { "Type" }
+                            th { "Default" }
+                            th { "Description" }
+                            th { "Additional Meta" }
+                        }}
+                        tbody class="border" {
+                            @for param in self.inputs_in_group(&group) {
+                                (param.render())
+                            }
+                        }
+                    }
+                }
+            });
+        html! {
+            @for group_table in group_tables {
+                (group_table)
+            }
+        }
+    }
+
+    /// Render the inputs that are neither required nor part of a group.
+    fn render_other_inputs(&self) -> Markup {
+        let mut iter = self.other_inputs().peekable();
+        if iter.peek().is_some() {
+            return html! {
+                h3 { "Other Inputs" }
+                table class="border" {
+                    thead class="border" { tr {
+                        th { "Name" }
+                        th { "Type" }
+                        th { "Default" }
+                        th { "Description" }
+                        th { "Additional Meta" }
+                    }}
+                    tbody class="border" {
+                        @for param in iter {
+                            (param.render())
+                        }
+                    }
+                }
+            };
+        };
+        html! {}
+    }
+
+    /// Render the inputs of the callable.
+    fn render_inputs(&self) -> Markup {
+        html! {
+            h2 { "Inputs" }
+            (self.render_required_inputs())
+            (self.render_common_inputs())
+            (self.render_group_inputs())
+            (self.render_other_inputs())
+        }
+    }
+
+    /// Render the outputs of the callable.
+    fn render_outputs(&self) -> Markup {
+        html! {
+            h2 { "Outputs" }
+            table  {
+                thead class="border" { tr {
+                    th { "Name" }
+                    th { "Type" }
+                    th { "Expression" }
+                    th { "Description" }
+                    th { "Additional Meta" }
+                }}
+                tbody class="border" {
+                    @for param in self.outputs() {
+                        (param.render())
+                    }
+                }
+            }
+        }
+    }
+}
+
+/// Parse a [`MetadataSection`] into a [`MetaMap`].
+fn parse_meta(meta: &MetadataSection) -> MetaMap {
```

Comment:
Can we process `meta` and `parameter_meta` together? Is there a generic trait or type that both share to enable a single function?

Reply:
I can play around with this locally and see if I can join these methods together. Good thought

Reply:
Actually this isn't so straightforward. I played around with it for a bit locally, and ran into other issues. I think we have to do it like this 🤷